cmd/tether/attach.go

```diff
@@ -255,12 +255,23 @@ func backchannel(ctx context.Context, conn *net.Conn) error {
 			if ctx.Err() != nil {
 				return ctx.Err()
 			}
-			err := serial.HandshakeServer(ctx, *conn)
-			if err != nil {
-				log.Error(err)
-			} else {
+			deadline, ok := ctx.Deadline()
+			if ok {
+				conn.SetReadDeadline(deadline)
```

Comment:
this has zero effect when backed by a rawconn - the underlying file Read has no timeout.
The only way to interrupt the call is by closing the fd.

Should likely change rawconn:SetReadDeadline, et al, to panic so there's no confusion now we have more people in this part of the code.